src/layers/fp64/scatterplot-layer/scatterplot-layer-64.js

```diff
@@ -18,96 +18,35 @@
-import {GL, Model, Geometry} from 'luma.gl';
 import {readFileSync} from 'fs';
 import {join} from 'path';
 
-const DEFAULT_COLOR = [255, 0, 255, 255];
+export default class ScatterplotLayer64 extends ScatterplotLayer {
 
-const defaultGetPosition = x => x.position;
-const defaultGetRadius = x => x.radius;
-const defaultGetColor = x => x.color || DEFAULT_COLOR;
-
-const defaultProps = {
-  getPosition: defaultGetPosition,
-  getRadius: defaultGetRadius,
-  getColor: defaultGetColor,
-  radius: 30,
-  radiusMinPixels: 0,
-  radiusMaxPixels: Number.MAX_SAFE_INTEGER,
-  drawOutline: false,
-  strokeWidth: 1
-};
-
-export default class ScatterplotLayer64 extends Layer {
-  constructor(props) {
-    super(Object.assign({}, defaultProps, props));
-  }
-
-  initializeState() {
-    const {gl} = this.context;
-    const {attributeManager} = this.state;
-
-    this.setState({
-      model: this.getModel(gl)
-    });
-
-    attributeManager.addInstanced({
-      instancePositionsFP64: {size: 4, update: this.calculateInstancePositions},
-      instanceRadius: {size: 1, update: this.calculateInstanceRadius},
-      layerHeight: {size: 2, update: this.calculateLayerHeight},
-      instanceColors: {
-        size: 4,
-        type: GL.UNSIGNED_BYTE,
-        update: this.calculateInstanceColors
-      }
-    });
-  }
-
-  draw({uniforms}) {
-    this.calculateZoomRadius();
-    this.state.model.render(Object.assign({}, uniforms, {
-      radiusMinPixels: this.props.radiusMinPixels,
-      radiusMaxPixels: this.props.radiusMaxPixels,
-      zoomRadiusFP64: this.state.zoomRadiusFP64
-    }));
-  }
-
-  getShaders() {
+  // Override the super class vertex shader
+  getShaders(id) {
     return {
       vs: readFileSync(join(__dirname, './scatterplot-layer-vertex.glsl'), 'utf8'),
-      fs: readFileSync(join(__dirname, './scatterplot-layer-fragment.glsl'), 'utf8'),
+      fs: super.getShaders().fs,
       fp64: true,
       project64: true
     };
   }
 
-  getModel(gl) {
-    const NUM_SEGMENTS = 16;
-    const positions = [];
-    for (let i = 0; i < NUM_SEGMENTS; i++) {
-      positions.push(
-        Math.cos(Math.PI * 2 * i / NUM_SEGMENTS),
-        Math.sin(Math.PI * 2 * i / NUM_SEGMENTS),
-        0
-      );
-    }
-
-    const shaders = assembleShaders(gl, this.getShaders());
+  initializeState() {
+    // We the model and several attributes of the base layer
+    super.initializeState();
 
-    return new Model({
-      gl,
-      id: this.props.id,
-      vs: shaders.vs,
-      fs: shaders.fs,
-      geometry: new Geometry({
-        drawMode: GL.TRIANGLE_FAN,
-        positions: new Float32Array(positions)
-      }),
-      isInstanced: true
+    // Add the 64 bit positions
+    const {attributeManager} = this.state;
+    attributeManager.addInstanced({
+      instancePositionsFP64: {size: 4, update: this.calculateInstancePositions},
+      instanceHeightsFP64: {size: 2, update: this.calculateInstanceHeightsFP64}
```

Comment:
> could you remind me why we need instanceHeights for?

Good point. I will rename these attributes toa `instancePositions64xy` and `instancePositions64z` that will hopefully make things clearer.